Source/Evolve/WarpXEvolveEM.cpp

```diff
@@ -88,6 +88,10 @@ WarpX::EvolveEM (int numsteps)
             UpdateAuxilaryData();
         }
 
+        // Performs continuous injection of all WarpXParticleContainer
+        // in mypc. 
+        mypc->ContinuousInjection(dt[0], geom[0].ProbDomain());
```

Comment:
As discussed off-line, this could be in the code that moves the window and generates new plasma, so that the `ContinuousInjection` is also implemented and used by the `PhysicalParticleContainer`.

Reply:
For the moment, the continuous injection for laser particles injects all particles at once. It assumes that the laser propagates in the direction of moving window (currently it has to be along z). So we can:
1/ generalize it so that it works the same for any moving window direction (i.e., moving window *and* laser propagation along *x* or *y* or *z*). That's easy.
2/ generalize it so that it works even if the laser direction differs from moving window direction. Probably more useful.
I suggest we go for solution 1 on this branch (where we add `PhysicalParticleContainer::ContinuousInjection`), and make a separate PR to this branch with solution 2. @RemiLehe what do you think?

Reply:
Agreed. Solution 1 is what we need for the vast majority of use cases.